tests/query/processors/test_mandatory_condition_applier.py

```diff
@@ -0,0 +1,51 @@
+import pytest
+
+
+# from snuba.clickhouse.columns import ColumnSet, String
+
+from snuba.datasets.schemas.tables import TableSource
+from snuba.query.logical import Query
+from snuba.query.processors.mandatory_condition_applier import MandatoryConditionApplier
+
+from snuba.request.request_settings import HTTPRequestSettings
+
+
+test_data = [
+    ("table1", [["time1", "=", "1"]]),
+    ("table2", [["time2", "=", "2"]]),
+]
+
+
+@pytest.mark.parametrize("table, mand_condition", test_data)
+def test_prewhere(table, mand_condition) -> None:
+
+    body = {
+        "conditions": [
+            ["d", "=", "1"],
+            ["c", "=", "3"],
+            ["a", "=", "1"],
+            ["b", "=", "2"],
+        ],
+    }
+
+    # cols = ColumnSet([("col", String())])
+
+    cols = None
+    consistent = True
+
+    query = Query(body, TableSource(table, cols, mand_condition, ["c1"]),)
+
+    request_settings = HTTPRequestSettings(consistent=consistent)
+    processor = MandatoryConditionApplier("sentry_dist", "sentry_ro")
+    processor.process_query(query, request_settings)
+
+    source = query.get_data_source()
+
+    assert isinstance(source, TableSource)
+    assert source.get_columns() == cols
+    assert source.get_prewhere_candidates() == ["c1"]
```

Comment:
I think you brought this over from the prewhere candidates. While they would work, these are not assertions you are really interested in verifying in this unit test since it is about testing that the mandatory conditions are. properly applied to your query.